### ISSUE TITLE
[JavaScript API] 'onAfterFormieValidate' event fetches wrong 'currentPageId' in it's event on multi-step form.

### ISSUE THREAD

**Issue by Antimated** [verbb/formie#775]
**Description**
Hi!

I'm trying to fetch the `currentPageId` in the 'onFormieValidate' event and manipulate some data on screen related to that specific step. The problem is that I always seem to get the same `currentPageId` (the one of the first page) back from the events `settings` object.
I trimmed down my code to just the specifics so I hope this helps.

```
const form = document.querySelector('.fui-form');

form.addEventListener('onAfterFormieValidate', (e) => {
    e.preventDefault();
    // Fetch the current page ID (is the same for every step for some reason, should not be the case!)
    const currentPageId = e.detail.submitHandler.settings.currentPageId ?? null;

    // The same
    console.log(currentPageId); 
    // Select the current step's html that is displayed
    const currentPageElement = form.querySelector(`[data-id="${currentPageId}"]`) ?? null;

    if(currentPageElement) {
        // Here I do an async call to some external method with a callback
        myMethodWithCallback(() => {
            e.detail.submitHandler.submitForm();
        });
        
    } else {
        // Instantly submit form
        e.detail.submitHandler.submitForm();
    }
});
```

**Steps to reproduce**
1. Create a multistep form (ajax enabled)
2. Use the javascript snippet above to target the form
3. Look at the `currentPageId` that is being printed in the console.

### Form settings
Please pick from the options below.

- Multi-page form: Yes
- Submission Method: Ajax
- Client-side Validation: Yes
- Custom Form Templates: Yes

**Additional info**
- Plugin version: 1.5.9
- Craft version: 3.7.33 (Pro)
- Multi-site: No

**Additional context**
Only thing that is being overwritten in the custom form templates is the submit button. Only displays some additional text next to the button.

**Comment by engram-design** [comment]
So the `settings` will be largely static, as they are populated only on page load to bootstrap the JS, which is why you're likely seeing them the same. What does get updated is the properties in the `formTheme` class.

`console.log(e.detail.submitHandler.formTheme.currentPageId); `

This is because the `formTheme` class is Formie's "themed" JS that handles page transitioning, ajax-calling, validation and a whole lot more of the UI stuff. As such, its correct that if you're opting-out of using our themed JS, you'll miss out, but it's largely up to you to implement all the JS. In that instance, you'd read the payload info from the submit controller (which would include the current/next page ids).

It could be argued that we update the `settings` object with the response back from Ajax calls, just to save confusion.

**Comment by Antimated** [comment]
Hmmm I'm not using the theme... any way to still get the currentPageId? Kind of mandatory in my implementation :(

**Comment by Antimated** [comment]
Or is the themed js the javascript that handles the displaying of errors?

Edit: would be nice if the settings object is updated appropriately, indeed kinda confusing that it's not updating 😃

**Comment by engram-design** [comment]
Yeah gotcha, as I mentioned, that's kinda the caveat here. But yes the themed JS handles client-side validation and lots more.

Will look into updating the settings object as the source of truth, instead of the theme class.

**Comment by engram-design** [comment]
Fixed for the next release. To get this fix early, change your `verbb/formie` requirement in `composer.json` to:

```
"require": {
  "verbb/formie": "dev-craft-3 as 1.5.9",
  "...": "..."
}
```

Then run `composer update`.

**Comment by engram-design** [comment]
Fixed in [1.5.10](https://github.com/verbb/formie/releases/tag/1.5.10)
